src/go/k8s/controllers/redpanda/redpanda_controller.go

```diff
@@ -425,8 +422,9 @@ func (r *RedpandaReconciler) createHelmReleaseFromTemplate(ctx context.Context,
 
 	return &helmv2beta1.HelmRelease{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      rp.GetHelmReleaseName(),
-			Namespace: rp.Namespace,
+			Name:            rp.GetHelmReleaseName(),
+			Namespace:       rp.Namespace,
+			OwnerReferences: []metav1.OwnerReference{rp.OwnerShipRefObj()},
```

Comment:
NIT:

Maybe you could use
```
err := controllerutil.SetControllerReference(rp, helmRelease, r.scheme)
```

REF:
https://github.com/redpanda-data/redpanda/blob/fcae600f00b178e9e1367af72d4e2a3263b7710a/src/go/k8s/pkg/resources/secret.go#L128